src/io/fileops.c

```diff
@@ -246,6 +246,35 @@ MVMint64 MVM_file_isexecutable(MVMThreadContext *tc, MVMString *filename, MVMint
     }
 }
 #else
+
+static int are_we_group_member(MVMThreadContext *tc, gid_t group) {
+    int len;
+    gid_t *gids = NULL;
+    int res;
+    int i;
+    // Check the user group.
+    if (getegid() == group)
+        return 1;
+    // Check the supplementary groups.
+    len  = getgroups(0, NULL);
+    if (len == 0)
+        return 0;
+    gids = MVM_realloc(gids, len * sizeof(gid_t));
+    res = getgroups(len, gids);
+    if (res < 0) {
+        MVM_free(gids);
+        MVM_exception_throw_adhoc(tc, "Failed to retrieve groups: %s", res);
```

Comment:
`res` is an `int`, so using it with `%s` will promptly segfault. I think you need to call `strerror(res)` to turn it into a string.

Reply:
Oh, sorry, I should read docs more carefully. It's actually `errno` that you need to pass to `strerror`. :-)